CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+# Contributuing
+
+Contributions to this project are welcome, whether minor updates to documentation or entirely new features.
+This document outlines the technical and procedural requirements for your work to be included.
+All contributions, in any form, must also confirm with the [code of conduct](CODE_OF_CONDUCT.md).
+
+> :warning: This document is a living document, and will adapt based on the project's needs.
+> If some part of this document creates a hindrance, raise an issue explaining the problem.
+
+## Scope: What Lives in this Repository?
+
+MewBot's core design includes the tenant of modularity.
+The core repo will only accept features and components that do not add additional dependencies.
+
+When building new components, whether they are input/output modules or utilities for use in behaviours,
+consider if they could be a stand-alone repository and package.
+The MewBot team is happy to help set up these repositories in the GitHub group.
+
+## Projects and Issues: What Needs Doing?
+
+Issues are a fantastic way of keeping track of what work is needed and who is planning to do it.
+You can find all the current initiatives in the projects tab; these show the current areas of focus.
+If you are looking for a way to contribute to the project, this is a great place to start.
+
+Issues are also a place to plan how to approach future work.
+Openly discussing ideas allows us to find places where multiple people are looking at
+similar problems and ensure that we don't have different solutions.
+
+When starting on a piece of work, check if there is a current issue you can assign to yourself.
+For more extensive changes, it is encouraged that you create it if one does not already exist.
+If another user has the issue assigned, coordinate your work with them.
+
+## Code Style
+
+Coding tackles the thorny problem of trying to express logical, computational concepts in a
+human-readable and understandable manner. Our code style favours the human side.
+We seek to find the balance between sections of code that are too large or dense to
+comprehend and the problem of too many small pieces of code to understand at once.
+
+Formatting is derived from Python's [PEP8](https://peps.python.org/pep-0008/) as implemented by
+[Black](https://pypi.org/project/black/), with line lengths limited to 100 (instead of the traditional 80).
+
+The linting toolchain uses
+[Black](https://pypi.org/project/black/),
+[Flake8](https://flake8.pycqa.org/en/latest/),
+[mypy](http://www.mypy-lang.org/), and
+[pylint](https://pylint.pycqa.org/en/latest/)
+to promote best practices for expressing the logical constructs.
+
+Expressing intent is the other key ingredient.
+Name code elements (classes, functions, variables) based on what purpose they serve,
+and try to organise code to convey the flow of logic.
+If there is additional information, use single-line comments.
```

Comment:
Suggest adding - just as a thought - and very happy to be argued off it. "More is better here in terms of clarity and content. If individual sections of code are becoming unwieldy then extensive comments may be moved into the wiki. If you do this, please include a link to the wiki page section where the comments now reside. Notes as to intent are also welcome."

Reply:
Ah - see the section below partially deals with it.

Reply:
Would also reccomend adding "Notes to intent should be removed prior to merge, or with a suitable issue raised as part of the pull request process"

Reply:
That's probably a better solution - that way people can go back and see the notes before the final pruning in the commits that made up the branch? Which is definitely more elegant than leaving them in the final code base.

We could even include links on the relevant wiki page to the last commit containing them.

Reply:
(Possidbly adding something about how 

"Extensive comments, or comments which should be read prior to working on the component, should be transfered into the wiki and a link to the relevant page included in the code")

Reply:
(But I am far from fixed on this, and the tone/content of the wiki is a discussion for another point.

Long quite liked the idea of a contributor wiki, where people can dump any notes they make during dev. Will add an issue).